packages/astro/src/runtime/server/astro-env.ts

```diff
@@ -0,0 +1,22 @@
+import type { GetEnv } from '../../env/types.js';
+import { AstroError, AstroErrorData } from '../../core/errors/index.js';
+export { validateEnvVariable } from '../../env/validators.js';
+
+let _getEnv: GetEnv = (key) => process.env[key];
```

Comment:
I also think that this should try by default if `setGetEnv` wasn't called. Otherwise there's a risk that it's called in non-node runtimes and makes it hard to debug if the env are working.

Reply:
That's actually a great catch. Maybe we shouldn't default to `process.env` at all, and let it define in the adapter regardless

Reply:
Well I thought about that and I think it makes sense to keep it as is by default. In particular for static sites because we want all astro sites, no matter the output, to be able to use any kind of variable. If we don't provide a default then people could not use it. Alternatively, we could default to `process.env` only when the output is static from within the virtual module

Reply:
Shouldn't all kinds of build be responsible for calling `setGetEnv` so that they all work? Which means for static builds, we can also call it manually before rendering the pages.